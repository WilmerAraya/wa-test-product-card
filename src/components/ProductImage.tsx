import React, { CSSProperties, useContext } from 'react';
import { ProductContext } from './ProductCard';

import noImage from '../assets/no-image.jpg';
import styles from '../styles/styles.module.css';

export interface Props {
  className?: string;
  img?: string;
  style?: CSSProperties;
}

export const ProductImage = ({ className, img, style }: Props) => {
  const { product } = useContext(ProductContext);
  const imgToShow: string = img || product.img || noImage;

  return (
    <img
      className={`${className} ${styles.productImg}`}
      style={style}
      src={imgToShow}
      alt="Coffee Mug"
    />
  );
};
