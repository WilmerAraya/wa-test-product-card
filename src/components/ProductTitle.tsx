import React, { CSSProperties, useContext } from 'react';

import { ProductContext } from './ProductCard';

import styles from '../styles/styles.module.css';

export interface Props {
  className?: string;
  title?: string;
  style?: CSSProperties;
}

// Otra forma de recibir las props
export const ProductTitle = ({ className, title, style }: Props) => {
  const { product } = useContext(ProductContext);
  const titleToShow: string = title || product.title;
  return (
    <span className={`${styles.productDescription} ${className}`} style={style}>
      {titleToShow}
    </span>
  );
};
