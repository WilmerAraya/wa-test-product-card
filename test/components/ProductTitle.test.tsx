import React from 'react';
import { render } from '@testing-library/react';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('debe mostrar el componente correctamente con una imagen personalizada', () => {
    // Renderizamos el componente ProductImage con una imagen personalizada
    const { baseElement } = render(<ProductImage img="https://hola.jpg" />);

    // Verificamos que el HTML renderizado coincide con el snapshot
    expect(baseElement.innerHTML).toMatchSnapshot();
  });

  test('debe mostrar el componente con la imagen del producto', () => {
    // Renderizamos ProductCard con el producto2 y dentro usamos ProductImage
    const { baseElement } = render(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    // Verificamos que el HTML renderizado coincide con el snapshot
    expect(baseElement.innerHTML).toMatchSnapshot();
  });
});
