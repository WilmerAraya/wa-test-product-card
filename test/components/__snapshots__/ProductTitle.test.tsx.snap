// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe mostrar el componente con la imagen del producto 1`] = `"<div><div class=\\"productCard undefined\\"><img class=\\"undefined productImg\\" src=\\"./coffee-mug.png\\" alt=\\"Coffee Mug\\"></div></div>"`;

exports[`ProductImage debe mostrar el componente correctamente con una imagen personalizada 1`] = `"<div><img class=\\"undefined productImg\\" src=\\"https://hola.jpg\\" alt=\\"Coffee Mug\\"></div>"`;
